src/helm/proxy/clients/lit_gpt_client.py

```diff
@@ -0,0 +1,149 @@
+                                              DecodeRequestResult,
+                                              TokenizationRequest,
+                                              TokenizationRequestResult,
+                                              TokenizationToken)
+
+from .client import Client, wrap_request_time
+from .lit_gpt_generate import generate
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+torch.set_float32_matmul_precision("high")
+
+
+class LitGPTClient(Client):
+    """Implements some "models" that just generate silly things quickly just to debug the infrastructure."""
+
+    def __init__(
+        self,
+        cache_config: CacheConfig,
+        checkpoint_dir: str = "",
+        precision: str = "bf16-true",
+        device="auto",
+        devices: int = 1,
+        strategy: str = "auto",
+        quantize: Optional[Literal["bnb.nf4", "bnb.nf4-dq", "bnb.fp4", "bnb.fp4-dq", "bnb.int8", "gptq.int4"]] = None,
+    ):
+        self.cache = Cache(cache_config)
+        if strategy == "fsdp":
+            strategy = FSDPStrategy(auto_wrap_policy={Block}, cpu_offload=False)
+        fabric = L.Fabric(devices=devices, accelerator=device, precision=precision, strategy=strategy)
+        fabric.launch()
+        logger.info("Using device: {}".format(fabric.device))
+
+        checkpoint_dir = Path(checkpoint_dir)
+        check_valid_checkpoint_dir(checkpoint_dir)
+
+        with open(checkpoint_dir / "lit_config.json") as fp:
+            config = Config(**json.load(fp))
+
+        checkpoint_path = checkpoint_dir / "lit_model.pth"
+        logger.info(f"Loading model {str(checkpoint_path)!r} with {config.__dict__}")
+        with fabric.init_module(empty_init=True), quantization(quantize):
+            model = GPT(config)
+
+        with lazy_load(checkpoint_path) as checkpoint:
+            model.load_state_dict(checkpoint, strict=quantize is None)
+
+        model.eval()
+        self.model = fabric.setup(model)
+        self.tokenizer = Tokenizer(checkpoint_dir)
+        self.fabric = fabric
+
+    def make_request(self, request: Request) -> RequestResult:
+        model = self.model
+        tokenizer = self.tokenizer
+        fabric = self.fabric
+        encoded = tokenizer.encode(
+            request.prompt, bos=True, eos=False, device=fabric.device
+        )
+        prompt_length = encoded.size(0)
+        max_returned_tokens = prompt_length + request.max_tokens
+        assert max_returned_tokens <= model.config.block_size, (
+            max_returned_tokens,
+            model.config.block_size,
+        )  # maximum rope cache length
+
+        model.reset_cache()
+        t0 = time.perf_counter()
+        tokens, logprobs, top_logprobs = generate(
+            model,
+            encoded,
+            max_returned_tokens,
+            max_seq_length=max_returned_tokens,
+            temperature=max(request.temperature, 1e-9),
+            top_k=request.top_p,
```

Comment:
`top_k` and `top_p` are not the same thing, so this is incorrect. The HELM request doesn't actually have anything that is equivalent to `top_k` currently.

We don't use `top_p` very much so we can just skip supporting it.

[OpenAI's explanation of `top_p`](https://platform.openai.com/docs/api-reference/completions):

> An alternative to sampling with temperature, called nucleus sampling, where the model considers the results of the tokens with top_p probability mass. So 0.1 means only the tokens comprising the top 10% probability mass are considered.

Reply:
removed `top_k`